bigchaindb/core.py

```diff
@@ -75,101 +71,25 @@ def reconnect(self):
-            operation (str): Either `CREATE` or `TRANSFER` operation.
-            payload (Optional[dict]): dictionary with information about asset.
-
-        Returns:
-            dict: unsigned transaction.
-
-
-        Raises:
-            TypeError: if the optional ``payload`` argument is not a ``dict``.
+        Refer to the documentation of ``bigchaindb.util.create_tx``
         """
-        data = None
-        if payload is not None:
-            if isinstance(payload, dict):
-                hash_payload = hash_data(self.serialize(payload))
-                data = {
-                    'hash': hash_payload,
-                    'payload': payload
-                }
-            else:
-                raise TypeError('`payload` must be an dict instance')
 
-        hash_payload = hash_data(self.serialize(payload))
-        data = {
-            'hash': hash_payload,
-            'payload': payload
-        }
-
-        tx = {
-            'current_owner': current_owner,
-            'new_owner': new_owner,
-            'input': tx_input,
-            'operation': operation,
-            'timestamp': self.timestamp(),
-            'data': data
-        }
-
-        # serialize and convert to bytes
-        tx_serialized = self.serialize(tx)
-        tx_hash = hash_data(tx_serialized)
-
-        # create the transaction
-        transaction = {
-            'id': tx_hash,
-            'transaction': tx
-        }
-
-        return transaction
+        return util.create_tx(current_owner, new_owner, tx_input, operation, payload)
 
     def sign_transaction(self, transaction, private_key):
         """Sign a transaction
 
-        A transaction signed with the `current_owner` corresponding private key.
-
-        Args:
-            transaction (dict): transaction to sign.
-            private_key (str): base58 encoded private key to create a signature of the transaction.
-
-        Returns:
-            dict: transaction with the `signature` field included.
-
+        Refer to the documentation of ``bigchaindb.util.sign_tx``
         """
-        private_key = PrivateKey(private_key)
-        signature = private_key.sign(self.serialize(transaction))
-        signed_transaction = transaction.copy()
-        signed_transaction.update({'signature': signature})
-        return signed_transaction
 
-    def verify_signature(self, signed_transaction):
-        """Verify the signature of a transaction
+        return util.sign_tx(transaction, private_key)
 
-        A valid transaction should have been signed `current_owner` corresponding private key.
-
-        Args:
-            signed_transaction (dict): a transaction with the `signature` included.
-
-        Returns:
-            bool: True if the signature is correct, False otherwise.
+    def verify_signature(self, signed_transaction):
+        """Verify the signature of a transaction.
 
+        Refer to the documentation of ``bigchaindb.crypto.verify_signature``
```

Comment:
This method is just a proxy to a function in `bigchaindb.crypto.verify_signature` (acutally, it's in `bigchaindb.util.verify_signature`) so I thought it would be nice to refer to the documentation to the original function.